analyzers/src/SonarAnalyzer.CFG/Roslyn/ControlFlowGraphCache.cs

```diff
@@ -33,33 +33,40 @@ public abstract class ControlFlowGraphCacheBase
 
         public ControlFlowGraph FindOrCreate(SyntaxNode declaration, SemanticModel model, CancellationToken cancel)
         {
-            var rootSyntax = model.GetOperation(declaration).RootOperation().Syntax;
-            var nodeCache = compilationCache.GetValue(model.Compilation, x => new());
-            if (!nodeCache.TryGetValue(rootSyntax, out var wrapper))
+            if (model.GetOperation(declaration) is { } declarationOperation)
```

Comment:
Please do this and write some UT for it. We will need it for sonar-security.
```suggestion
var cfgNode = declaration switch 
{
    IndexerDeclarationSyntax indexer => indexer.ExpressionBody,
    PropertyDeclarationSyntax property => property.ExpressionBody,
    _ => declaration
};
            if (model.GetOperation(cfgNode) is { } declarationOperation)
```

Reply:
@pavel-mikula-sonarsource I'm doing this [in the runner here](https://github.com/SonarSource/sonar-dotnet/pull/8073/files#diff-7c2044a5e7097e4323205fc4ee570cc0d1bae3c95422ede523982ec15bd14ef4R89),

Just to be clear: the comment is to transfer the logic here right? Not to add it.

Reply:
Yes